internal/provider/kubernetes/resources/statefulset.go

```diff
@@ -146,6 +153,18 @@ func (s *StatefulSetWithOrphanDelete) Delete(ctx context.Context) error {
 	if k8serrors.IsNotFound(err) {
 		return errors.NewNotFound(err, "k8s statefulset for deletion")
 	}
+	// K8s doesn't delete the sts immediately. Block until it's deleted.
+	err = wait.PollUntilContextTimeout(ctx, s.interval, s.timeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
nit: we don't seem to be using named returned params here, i.e done bool, err error

we can edit these to be (bool, err)